### PR TITLE
DEV: make make_release.py compatible with windows environment

### PR DESCRIPTION
**Pull request by pubpub-zz**

### CODE REVIEW

**Comment by pubpub-zz** [comment]
the doc compilation failed randomly. This should be OK

File: make_release.py [hunk 1]
```diff
@@ -267,7 +269,7 @@ def get_most_recent_git_tag() -> str:
     """
     git_tag = str(
         subprocess.check_output(
-            ["git", "describe", "--abbrev=0"], stderr=subprocess.STDOUT
+            ["git", "describe", "--tag", "--abbrev=0"], stderr=subprocess.STDOUT
```
Comment (stefan6419846) [inline make_release.py:272]: What version of `git` are you using?

File: make_release.py [hunk 2]
```diff
@@ -349,10 +355,8 @@ def parse_commit_line(line: str, authors: Dict[str, str]) -> Change:
 
     # Standardize
     message.strip()
-    commit_hash = commit_hash.strip('"')
+    commit_hash = commit_hash.strip('"\\ \r\n')
```
Comment (stefan6419846) [inline make_release.py:358]: Do you have an example for this? What happens if we use `commit_hash.strip().strip('"')`  instead? Especially the backslash looks odd for a commit hash.

Comment (pubpub-zz) [reply make_release.py:358]: I uses git version 2.37.2 (msys64)

File: make_release.py (hunk 1 above)
Comment (pubpub-zz) [reply make_release.py:272]: I uses git version 2.37.2 (msys64)

File: make_release.py (hunk 2 above)
Comment (pubpub-zz) [reply make_release.py:358]: My issue was with \\ I just completed the clean-up

Comment (stefan6419846) [reply make_release.py:358]: Could you please provide some example for this and add a corresponding test then?

`commit_hash.strip()` makes sure to strip all whitespace, thus, I would propose to use

```suggestion
    commit_hash = commit_hash.strip().strip('"\\')
```

here.

**Comment by stefan6419846** [comment]
Could you please add a test to cover the `parse_commit_line` change accordingly?

**Comment by pubpub-zz** [comment]
I have no client on how to test a program file not within tests folder

**Comment by stefan6419846** [comment]
There is nothing special about this, see the existing test: https://github.com/py-pdf/pypdf/blob/main/tests/scripts/test_make_release.py

**Review by MartinThoma — DISMISSED** [review]
I just gave it a shot - works for me

(except `parse_commit_line` where I had to replace line 360 by `author_login = authors.get(commit_hash, author)`, but that is not related to the PR)

**Comment by pubpub-zz** [comment]
> I just gave it a shot - works for me
> 
> (except `parse_commit_line` where I had to replace line 360 by `author_login = authors.get(commit_hash, author)`, but that is not related to the PR)

Can you clarify what error did you get and what was the commit_hash value ?

**Comment by stefan6419846** [comment]
I would still prefer to see a test added here.

> (except `parse_commit_line` where I had to replace line 360 by `author_login = authors.get(commit_hash, author)`, but that is not related to the PR)

Which indicates that an author could not be mapped. In theory, this should not occur, although there might be edge cases. Nevertheless, this should probably be fixed and tested as well.

**Comment by pubpub-zz** [comment]
> I would still prefer to see a test added here.

Agréé. in progress


> Which indicates that an author could not be mapped. In theory, this should not occur, although there might be edge cases. Nevertheless, this should probably be fixed and tested as well.

Agree. I had a similar issue this is why I completed the strip() waiting for feedback from @MartinThoma

**Comment by MartinThoma** [comment]
It was a key-not-found error - the the hash was not in the dictionary. I haven't looked more into it.

**Comment by pubpub-zz** [comment]
@MartinThoma 
I've been able to reproduce the key not found error, running the script from a branch different to main. Can you confirm it was the same for you

**Comment by stefan6419846** [comment]
> I've been able to reproduce the key not found error, running the script from a branch different to main.

This is not surprising as this script is only intended for the default branch. Running on other branches would require different GitHub API requests.

**Comment by pubpub-zz** [comment]
> This is not surprising as this script is only intended for the default branch. Running on other branches would require different GitHub API requests.

I agree this is not the objective of this script. I've added the test based on what I've internally objerved in windows. Ready for review

**Comment by stefan6419846** [comment]
While the Windows CI breaks because of broken test file downloads, merging is not possible for now. We should probably wait some time before re-running it to avoid rate limit errors.

**Comment by pubpub-zz** [comment]
I never understood why the files seem to be properly cached in ubuntu but not in windows any ideas?

**Comment by stefan6419846** [comment]
Windows does not use the cache as it is not configured there.

**Comment by pubpub-zz** [comment]
> Windows does not use the cache as it is not configured there.

shouldn't we set it up ?

**Comment by stefan6419846** [comment]
At the moment, this helps detecting invalid links which are not available any more. Thus I am not completely sure about this.

**Review by stefan6419846 — APPROVED** [review]
